state/aws/dynamodb/dynamodb.go

```diff
@@ -74,25 +75,48 @@ func NewDynamoDBStateStore(_ logger.Logger) state.Store {
 }
 
 // Init does metadata and connection parsing.
-func (d *StateStore) Init(_ context.Context, metadata state.Metadata) error {
+func (d *StateStore) Init(ctx context.Context, metadata state.Metadata) error {
 	meta, err := d.getDynamoDBMetadata(metadata)
 	if err != nil {
 		return err
 	}
 
-	client, err := d.getClient(meta)
-	if err != nil {
-		return err
+	// We have this check because we need to set the client to  a mock in tests
+	if d.client == nil {
+		client, err := d.getClient(meta)
+		if err != nil {
+			return err
+		}
+		d.client = client
 	}
-
-	d.client = client
 	d.table = meta.Table
 	d.ttlAttributeName = meta.TTLAttributeName
 	d.partitionKey = meta.PartitionKey
 
+	if err := d.validateTableAccess(ctx); err != nil {
+		return fmt.Errorf("error validating DynamoDB table '%s' access: %w", d.table, err)
+	}
+
 	return nil
 }
 
+// validateConnection runs a dummy Get operation to validate the connection credentials,
+// as well as validating that the table exists, and we have access to it
+func (d *StateStore) validateTableAccess(ctx context.Context) error {
+	input := &dynamodb.GetItemInput{
+		ConsistentRead: aws.Bool(false),
+		TableName:      aws.String(d.table),
+		Key: map[string]*dynamodb.AttributeValue{
+			d.partitionKey: {
+				S: aws.String(uuid.NewString()),
```

Comment:
`uuid.NewString` can panic (if the kernel doesn't have entropy). Please use `uuid.NewRandom` and then call `.String()` on the result

Reply:
Also, ptr.Of

Reply:
Thanks @ItalyPaleAle! Check out if you like how I did this. In case of error, I chose to use a fixed string, since I didn't want to have the initialisation of dapr fail because of it; it's just a dummy check anyway.